### DISCUSSION TITLE
Can't purge cache with hard refresh when using "use cache"

### DISCUSSION THREAD

**Discussion by mordechaim** [vercel/next.js#72044] [Help]
### Link to the code that reproduces this issue

https://github.com/mordechaim/next-repro-cache-refresh

### To Reproduce

1. Start the application in dev mode
2. Observe the dates being cached in 2 different modes
3. Open devtools
4. Right click the browser refresh button
5. Choose "Empty Cache and Hard Reload"

### Current vs. Expected behavior

Hard reload allows us to purge the cache in development which is a crucial feature when making changes to a cached function.

This feature currently only works for the legacy `unstable_cache` but when using `"use cache"` the data remains cached.

**Expected**
The cache is cleared and fresh data is loaded

**Current**
Only the `unstable_cache` is purged, not `"use cache"`

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 10 Home
  Available memory (MB): 32674
  Available CPU cores: 8
Binaries:
  Node: 22.9.0
  npm: N/A
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 15.0.2-canary.9 // Latest available version is detected (15.0.2-canary.9).
  eslint-config-next: N/A
  react: 19.0.0-rc-1631855f-20241023
  react-dom: 19.0.0-rc-1631855f-20241023
  typescript: 5.3.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Developer Experience

### Which stage(s) are affected? (Select all that apply)

next dev (local)

**Comment by samcx** [comment]
@mordechaim I think this is by design and you need to use `cacheLife()` or `cacheTag()`, but let me confirm!

**Reply by mordechaim** [reply]
Sure, that's for real cache invalidation, but that's not ergonomic when purging the cache while developing to "just see the change I just made"

**Reply by samcx** [reply]
@mordechaim It's currently the expected behavior. Our team is looking to update this since it is indeed cumbersome!

Thank you for flagging it 🙇🏼

**Reply by leo-cheron** [reply]
Is it really "expected" not being able to flush the cache while developing? How are we supposed to work?

**Comment by aniruddhaadak80** [comment]
Hey @mordechaim ,

 Top three possible solutions to address the cache issue with `"use cache"` in your Next.js application:

1. **Disable Caching in DevTools**:
   - Open DevTools (F12).
   - Go to the **Network** tab.
   - Check **Disable cache** while DevTools is open.

2. **Modify Cache Key**: Change the cache key dynamically to ensure fresh data is fetched instead of relying on cached responses.

3. **Use HTTP Headers**: Set `Cache-Control: no-store` on your fetch requests to prevent caching during development.

Thanks .

**Reply by mordechaim** [reply]
Thank you ChatGPT, but non of those answer my question.

We are dealing with a _server_ cache.
1. The **Disable cache** function disables client-side cache by the browser.
2. In `"use cache"` there's no cache key
3. I'm caching an async function, not a fetch call.
